packages/did/src/did.ts

```diff
@@ -36,6 +39,39 @@ export const decodeFromDid = (tangleid: Did): MnidModel => {
   return mamParsed;
 };
 
+/**
+ * Checks if given URL is a DID URL.
+ * @function isDidUrl
+ * @param {string} url - the URL to be checked.
+ * @return {boolean} The boolean that represent the url whether DID URL.
+ */
+export const isDidUrl = (url: string): boolean => {
+  return DID_URL_REGEX.test(url);
+};
+
+/**
+ * Decode information of the DID URL.
+ * @function decodeFromDidUrl
+ * @param {string} url - the DID URL that will be decoded.
+ * @return {{did: string, method: string, idstring: string, fragment: string}}
+ *   The information of the DID URL. {@link https://w3c-ccg.github.io/did-spec/#generic-did-syntax}
+ */
+export const decodeFromDidUrl = (url: string) => {
+  const matches = DID_URL_REGEX.exec(url);
+  if (matches === null) {
+    throw new Error('Invalid DID URL');
+  }
+
+  const [, method, idstring, fragment] = matches;
```

Comment:
I wonder if there are restrictions over `idstring`, such as case sensitivity.

Reply:
According to [Generic DID Syntax](https://w3c-ccg.github.io/did-spec/#generic-did-syntax), the restrictions are:
 - `idstring`: `A-Za-z0-9.-_`
 - `method`: `a-z0-9`

I rewrote the regular expression and make it more strictly.